config/config-mp/src/main/java/io/helidon/config/mp/spi/MpMetaConfigProvider.java

```diff
@@ -44,4 +45,12 @@ public interface MpMetaConfigProvider {
      * @return list of config sources
      */
     List<? extends ConfigSource> create(String type, Config metaConfig, String profile);
+
+    /**
+     * Create the {@link ConfigSource} from the given content.
+     *
+     * @param content a reader with the content data
+     * @return config source
+     */
+    ConfigSource create(Reader content);
```

Comment:
Maybe a default implementation that throws an `UnsupportedOperationException` would be better here.
Also this would be a backward incompatible change.